impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/FilterApplicator.java

```diff
@@ -47,8 +49,10 @@ private FilterApplicator(Type type) {
     }
 
     public static FilterApplicator with(Type type, Filter filter) {
+        // note: this is a static method so this can't be achieved elegantly with polymorphism
```

Comment:
Oh I missed the fact that `Filter` is in the "API" and `FilterApplicator` is in the "API Impl". I don't know, what about something like this:

http://pastebin.test.redhat.com/268206

Far from perfect, though. It has nothing to do with polymorphism, but it's good to have all the binding at one place. Perhaps the visit method could be implemented in those applicator sub-classes as well.